### ISSUE TITLE
Set an index for a field

### ISSUE THREAD

**Issue by RoxasShadow** [benashford/rs-es#11]
Hello,

I was wondering how I can set an index for a certain field. Basically I have an array of strings `work_roles` that appears in the following way

```json
"work_roles": {
    "type": "string"
}
```

and my goal is to make it in this way:

```json
"work_roles": {
    "type": "string",
    "index": "not_analyzed"
}
```

Unfortunately I wasn't able to find anything in the docs. Am I missing something?

I guess it should be something like [this](https://stackoverflow.com/questions/22770391/error-on-creation-of-a-mapping-for-an-index).

**Comment by benashford** [comment]
No, not you are not missing anything, this is an as-yet unimplemented feature.

Eventually there will be an implementation for the [Indices API](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices.html) at the same level as the Search API, Bulk API, etc. (e.g. `src/operations/indices.rs`), but there are a few other tasks on the list before I start with this one.

In the meantime, since index creation is usually a one-time task, I'd recommend scripting it using one of the many libraries for languages like Python or Ruby.  For ongoing index management, Elastic have a tool called [Curator](https://github.com/elastic/curator) which can be handy for closing/aliasing and other such things.

But I do plan on adding that API to `rs-es` eventually, as it would be good to be able to create custom tools for managing index in Rust.

**Comment by RoxasShadow** [comment]
I ended up writing a very basic implementation for the index fields mapping, that is basically just what I currently need. I hope it's fine as first iteration.

**Comment by benashford** [comment]
Cool!  I'll probably move this into the larger implementation of the Indices API when I begin on that, but that won't be for another couple of versions at least.  This is a good start on that path.

Thanks.

**Comment by benashford** [comment]
Merged into master pending a 0.3.0 release.
